libtransmission/rpc-server.cc

```diff
@@ -643,6 +643,58 @@ void rpc_server_start_retry_cancel(tr_rpc_server* server)
     server->start_retry_counter = 0;
 }
 
+int tr_evhttp_bind_socket(struct evhttp* httpd, char const* address, ev_uint16_t port)
+{
+#ifdef _WIN32
```

Comment:
Is there a way to check for libevent version (even if planned/unreleased) that introduces the built-in support, so that we know when to remove this?

Reply:
I looked into the libevent code.

We have to have some custom if we want to achieve this goal of this PR. We can't directly use `evhttp_bind_socket()`, because in the path of, the socket options are hard coded.

https://github.com/libevent/libevent/blob/7f23b5b5aa61f5496627c2651fd6d8185f34ffde/http.c#L3914-L3915

What we can do instead, once 2.2.1 is released, is let libevent to do the hard work of creating the socket and error checking, something like this:

```c++
// ... getaddrinfo and whatnot

struct evconnlistener* lev = evconnlistener_new_bind(
    base,
    nullptr,
    nullptr,
    LEV_OPT_REUSEABLE | LEV_OPT_CLOSE_ON_EXEC | LEV_OPT_CLOSE_ON_FREE | LEV_OPT_BIND_IPV4_AND_IPV6,
    -1,
    reinterpret_cast<sockaddr const*>(&addr),
    sizeof(addr));

// ...

return evhttp_bind_listener(httpd, lev) != nullptr;
```